example/loop_advection/loop_advection_package.cpp

```diff
@@ -0,0 +1,368 @@
+  m = Metadata({Metadata::Edge, Metadata::Derived, Metadata::FillGhost},
+               scalar_shape);
+  package->AddField(E_edge::name(), m);
+
+  // The divergence of the face centered magnetic field at cell centers
+  m = Metadata({Metadata::Cell, Metadata::Derived});
+  package->AddField(div_Bf_cell::name(), m);
+
+  // The divergence of the cell centered magnetic field at cell centers
+  m = Metadata({Metadata::Cell, Metadata::Derived});
+  package->AddField(div_Bc_cell::name(), m);
+
+  // The divergence of the electric_field at nodes
+  m = Metadata({Metadata::Node, Metadata::Derived});
+  package->AddField(div_E_node::name(), m);
+
+  packages.Add(package);
+  return packages;
+}
+
+void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
+  using parthenon::MetadataFlag;
+
+  auto &md = pmb->meshblock_data.Get();
+
+  auto pkg = pmb->packages.Get("loop_advection_package");
+  const auto &amp = pkg->Param<Real>("amp");
+  const auto &R = pkg->Param<Real>("R");
+  const auto &v1 = pkg->Param<Real>("v1");
+  const auto &v2 = pkg->Param<Real>("v2");
+
+
+  auto &v =  SparsePack<E_edge,B_face>::Make(pmd->meshblock_data.Get());
+
+  using A_face = E_edge;
+
+  //Initialize vector potential at cell edges
+  const auto &A = md->PackVars(std::vector<std::string>{"E_edge"});
+  for( int edge_dir=1; edge_dir <= 3; edge_dir++){
+    CartDir edge_dir_enum = static_cast<CartDir>(edge_dir);
+
+    IndexRange ib = pmb->cellbounds.GetEdgeBoundsI(IndexDomain::entire,edge_dir_enum);
+    IndexRange jb = pmb->cellbounds.GetEdgeBoundsJ(IndexDomain::entire,edge_dir_enum);
+    IndexRange kb = pmb->cellbounds.GetEdgeBoundsK(IndexDomain::entire,edge_dir_enum);
+
+    pmb->par_for(
+      "LoopAdvection::ProblemGenerator::init_A.x"+(edge_dir)+"e", 0, vf.dim(5)-1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        const auto &coords = v.GetCoords(b);
+        const Real r = sqrt( x*x  + y*y + z*z );
+
+
+        const Real x = (edge_dir == 1) ? coords.x1v(i) : coords.x1f(i);
+        const Real y = (edge_dir == 2) ? coords.x2v(j) : coords.x2f(j);
+        const Real z = (edge_dir == 3) ? coords.x3v(k) : coords.x2f(k);
+
+        const Real ax = 0;
+        const Real ay = 0;
+        const Real az = max( amp*(R - r),0);
+
+        v( edge_dir, b, A_face(), k, j, i) = (edge_dir == 1 ) ? ax :
+                                             (edge_dir == 2 ) ? ay : az;
+      });
+  }
+
+
+  //Initialize magnetic field from curl of vector potential
+  //Computing each direction individually (not necessarily the fastest method)
+  for( int face_dir=1; face_dir <= 3; face_dir++){
+    CartDir face_dir_enum = static_cast<CartDir>(face_dir);
+
+    IndexRange ib = pmb->cellbounds.GetFaceBoundsI(IndexDomain::entire,face_dir_enum);
+    IndexRange jb = pmb->cellbounds.GetFaceBoundsJ(IndexDomain::entire,face_dir_enum);
+    IndexRange kb = pmb->cellbounds.GetFaceBoundsK(IndexDomain::entire,face_dir_enum);
```

Comment:
Yes, that is the correct math that will always get called. It's a question of whether we want that `+ (dir == face_dir)` inside these bounds functions or outside when the loops get called. I have a slight preference for putting that logic in one place rather than everywhere it's needed.

Reply:
Another option is to change to `GetBounds*(const IndexDomain &idx_domain, const CellElement& element = CellElement::Center);` with an enum
```c++
enum class CellElement {Center, Face_x, Face_y, Face_z, Edge_xy, Edge_xz, Edge_yz, Vertex};    
```

Reply:
I like this idea the most -- I'll implement this